src/Transaction.ts

```diff
@@ -166,9 +200,18 @@ export class Transaction {
     }
 
     return {
-      gasPrice: utils.formatToAmount(txFees.gasPrice, feeDecimals),
-      gasLimit: utils.formatToAmount(txFees.gasLimit, 0),
-      baseFee: utils.formatToAmount(txFees.baseFee, feeDecimals),
+      gasPrice:
+        txFees.gasPrice !== undefined
+          ? utils.formatToAmount(txFees.gasPrice, feeDecimals)
+          : null,
```

Comment:
I would not do such a change for `utils.formatToAmount` is used in so many places, and here the reason we return `null` is afaik to make it clear to the user that we did not forget to set the gasPrice (which would be undefined otherwise), while it could be that in some other places we actually wanted to return undefined.